### PR TITLE
fix(metrics): Count all users, not just init users [INGEST-1330]

### PR DESCRIPTION
**Pull request by jjbayer**
The javascript SDK apparently does not always send a session.distinct_id with the initial session update, but only on subsequent updates.

To make sure we count those users as well, Relay now extracts them under the tag session.status=ok (See also https://github.com/getsentry/relay/pull/1275).

This PR changes existing metrics user queries in one of two ways:

1. Wherever we filtered users by `status=init`, remove the filter.
2. Wherever we grouped by `status` to get multiple user counts, change the query to a conditional aggregate (`uniqIf`), so that we get both the total user count (unfiltered) _and_ the user counts filtered to specific tags.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_organization_metric_data.py [hunk 1]
```diff
@@ -2079,8 +2079,6 @@ def test_errored_user_sessions_clamped_to_zero(self):
     def test_healthy_user_sessions(self):
```
Comment (andriisoldatenko) [inline tests/sentry/api/endpoints/test_organization_metric_data.py:2079]: Code LGTM, it would be nice to add more tests around changes were made.

**Review by andriisoldatenko — CHANGES_REQUESTED** [review]

File: src/sentry/sentry_metrics/indexer/strings.py [hunk 2]
```diff
@@ -84,7 +84,7 @@
     "browser.name": PREFIX + 216,
     "os.name": PREFIX + 217,
     "satisfaction": PREFIX + 218,
-    "ok": PREFIX + 219,
+    "ok": PREFIX + 219,  # now also used by release health
```
Comment (untitaker) [inline src/sentry/sentry_metrics/indexer/strings.py:87]: i was assuming we'd extract this extra data under `exited`, no?

**Review by untitaker — APPROVED** [review]

File: tests/sentry/api/endpoints/test_organization_metric_data.py (hunk 1 above)
Comment (untitaker) [reply tests/sentry/api/endpoints/test_organization_metric_data.py:2079]: i agree. @jjbayer I also realized, you need to update metrics extraction in sentry's testharness

Comment (jjbayer) [reply tests/sentry/api/endpoints/test_organization_metric_data.py:2079]: > you need to update metrics extraction in sentry's testharness
Yep, working on it

File: src/sentry/sentry_metrics/indexer/strings.py (hunk 2 above)
Comment (jjbayer) [reply src/sentry/sentry_metrics/indexer/strings.py:87]: We emit the user ID with the `ok` tag not just for exited sessions, but for any session update that is not init, but still healthy (https://github.com/getsentry/relay/blob/243e85292d785a1010256eb6b2e1748191ec6708/relay-server/src/metrics_extraction/sessions.rs#L205-L218).

The reason for this is that not all sessions exit formally AFAIK, but we still want to capture the user ID when they don't.

**Review by andriisoldatenko — APPROVED** [review]

**Review by ahmedetefy — COMMENTED** [review]
Can you please fix all tests that use `init` session status for users by removing that tag 
like https://github.com/getsentry/sentry/blob/9b8a7d5e86adac46cc9eb0208bceb8d9bae1f5d4/tests/sentry/api/endpoints/test_organization_metric_data.py#L1697

**Pull request merged**